tests/MaxMind/Test/WebService/ClientTest.php

```diff
@@ -66,14 +101,16 @@ public function test200WithInvalidJson(): void
         $this->expectException(\MaxMind\Exception\WebServiceException::class);
         $this->expectExceptionMessage('Received a 200 response for TestService but could not decode the response as JSON: Syntax error. Body: {');
 
-        $this->withResponse(200, 'application/json', '{');
+        $this->withResponseTestServer(200, 'application/json', '{');
     }
 
     public function test204WithResponseBody(): void
     {
         $this->expectException(\MaxMind\Exception\WebServiceException::class);
         $this->expectExceptionMessage('Received a 204 response for TestService along with an unexpected HTTP body: non-empty response body');
 
+        // Fatih: I was not able to use the built-in server to send a body with 204. For some reason,
+        // the echo in the router.php doesn't send the body when the status is 204 (No-content).
         $this->withResponse(204, 'application/json', 'non-empty response body');
```

Comment:
I had to mock this response using the old way. I couldn't figure out why I was not able to send the body in the response of my test server when I set the http status to 204.